src/provider/model/AppDirectory.ts

```diff
@@ -16,6 +16,44 @@ enum StorageKeys {
 
 @injectable()
 export class AppDirectory extends AsyncInit {
+    /**
+     * Test if an app *might* support an intent - i.e., were the app running with the appropriate listener added,
+     * should we regard it as supporting this intent (and optionally context)
+     */
+    public static mightAppSupportIntent(app: Application, intentType: string, contextType?: string): boolean {
+        if (contextType === undefined || app.intents === undefined) {
+            return true;
+        } else {
+            const intent = app.intents.find(intent => intent.name === intentType);
+            return intent === undefined || intentSupportsContext(intent, contextType);
+        }
+    }
+
+    /**
+     * Test if an app *should* support an intent - i.e., were the app running, would we expect it to add a listener for
+     * the given intent, and would we then regard it as supporting this intent (and optionally context)
+     */
+    public static shouldAppSupportIntent(app: Application, intentType: string, contextType?: string): boolean {
+        if (app.intents === undefined) {
+            return false;
+        } else {
+            const intent = app.intents.find(intent => intent.name === intentType);
+            return intent !== undefined && (contextType === undefined || intentSupportsContext(intent, contextType));
+        }
+    }
+
+    public static getIntentDisplayName(apps: Application[], intentType: string): string {
+        for (const app of apps) {
+            const intent = app.intents && app.intents.find(intent => intent.name === intentType);
```

Comment:
I imagine there was some discussion that I missed around how we handle the fact that there's multiple names, but is "return the first from some arbitrarily sorted list" really the right way? I can't think of any better way to be honest, but still....

Reply:
There wasn't any real discussion around this, but I think this is what we were doing before somewhere, and it seemed like an improvement to just do that everywhere, rather than use the intent type for the display name.

We have https://appoji.jira.com/browse/SERVICE-751 if there's further improvements to be made

Reply:
Ok, that's fine by me.